### PR TITLE
chore: custom configuration for formatter tests

### LINKED ISSUE #3181: custom configuration for formatter tests
Currently, our code formatter tests have a default configuration, but there are cases where custom configurations are needed for specific scenarios.

We propose adding a syntax to specify custom configurations using comments in the code. The following example syntax can be used:
```rust
//@ remove_nested_parens = false
//@ error_on_unformatted = true
```

### PR DESCRIPTION
**Pull request by f01dab1e**
# Description

chore: custom configuration for formatter fests

## Problem

Resolves #3181 

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by f01dab1e** [comment]
@kevaundray

File: tooling/nargo_fmt/src/config.rs [hunk 1]
```diff
@@ -44,20 +44,25 @@ macro_rules! config {
 config! {
     tab_spaces: usize, 4, "Number of spaces per tab";
     remove_nested_parens: bool, true, "Remove nested parens";
+    error_on_unformatted: bool, false, "Error if unable to get comments";
```
Comment (jfecher) [inline tooling/nargo_fmt/src/config.rs:47]: The comment here doesn't match up with the flag name. From the flag name, it sounds like a `-Werror` equivalent but the comment suggests it has something to do with comments.

Comment (f01dab1e) [reply tooling/nargo_fmt/src/config.rs:47]: something like `error_on_lost_comment`?

Comment (jfecher) [reply tooling/nargo_fmt/src/config.rs:47]: Sure - and shouldn't a lost comment be an error by default?

Comment (f01dab1e) [reply tooling/nargo_fmt/src/config.rs:47]: by default, we replace it with the original expression; perhaps we should change this behavior

Comment (jfecher) [reply tooling/nargo_fmt/src/config.rs]: I don't think we should ever ideally lose or rewrite a user's comment so if it does happen I think an error would be preferable

**Comment by jfecher** [comment]
@f01dab1e can you fix the merge conflicts?

**Comment by f01dab1e** [comment]
@jfecher

**Comment by kevaundray** [comment]
fyi there is a failing formatter test here

**Comment by f01dab1e** [comment]
@jfecher

**Review by kevaundray — APPROVED** [review]

**Pull request merged**